src/Workspaces/Core/Portable/Workspace/Solution/SolutionState_Checksum.cs

```diff
@@ -133,14 +133,17 @@ public async Task<Checksum> GetChecksumAsync(ProjectId projectId, CancellationTo
                 var projectChecksums = allResults.SelectAsArray(r => r.Checksum);
                 var projectIds = allResults.SelectAsArray(r => r.ProjectId);
 
-                var analyzerReferenceChecksums = ChecksumCache.GetOrCreateChecksumCollection(
-                    this.AnalyzerReferences, this.Services.GetRequiredService<ISerializerService>(), cancellationToken);
+                var serializer = this.Services.GetRequiredService<ISerializerService>();
+
+                var analyzerReferenceChecksums = ChecksumCache.GetOrCreateChecksumCollection(AnalyzerReferences, serializer, cancellationToken);
+                var fallbackAnalyzerOptionsChecksum = serializer.CreateChecksum(FallbackAnalyzerOptions, cancellationToken);
```

Comment:
note: i think you want to go through ChecksumCache here, so that if we have hte same dict here, we get teh same checksum back immediately.  Otehrwise, i think this computes the whole serialized blob each time (which can be not cheap if these fallback options are large).